### PR TITLE
Sm/mpd

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
breaking change: SourceTracking.localChangesAsComponentSet returns an array of componentSets
feat: support for MPD where each pkgDir becomes a component set
does not return CS when there are no files in it (because no local changes, or no local changes that aren't hit by .forceignore)
NUT for MPD and non-MPD on `localChangesAsComponentSet`

--other non-related changes
- remove a typeguard that could have used ts-types
- remote virtualTree code that's now part of SDR

### What issues does this PR fix or reference?
Part 1 of @W-10146239@

### CODE REVIEW

File: src/sourceTracking.ts [hunk 1]
```diff
@@ -76,47 +76,77 @@ export class SourceTracking extends AsyncCreatable {
...
-  public async localChangesAsComponentSet(): Promise<ComponentSet> {
+  /**
+   *
+   * @param byPackageDir if true, returns one ComponentSet for each packageDir with changes
+   * @returns ComponentSet[]
+   */
+  public async localChangesAsComponentSet(byPackageDir = false): Promise<ComponentSet[]> {
     const [projectConfig] = await Promise.all([this.project.resolveProjectConfig(), this.ensureLocalTracking()]);
-    const sourceApiVersion = getString(projectConfig, 'sourceApiVersion');
+    // we don't "cache" this one in a member property since w
```
Comment (WillieRuemmele) [inline src/sourceTracking.ts:86]: is this comment complete?

File: test/nuts/mpd.nut.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  it('byPkgDir => 2 component when one pkgDir has no non-ignored files', async () => {
+    // will forceignore an entire directory--effectively, a pkgDir with no files found
+    const forceIgnoreLocation = path.join(session.project.dir, '.forceignore');
+    await fs.writeFile(forceIgnoreLocation, 'my-app/*');
+
+    // new instance of STL since we changed the forceignore (it'd be cached from previous tests)
+    stl = await getSTLInstance(session);
+    const cs = await stl.localChangesAsComponentSet(true);
+    // eslint-disable-next-line no-console
```
Comment (WillieRuemmele) [inline test/nuts/mpd.nut.ts:56]: not required anymore

**Review by WillieRuemmele — APPROVED** [review]
2 _VERY_ minor NITs

File: src/sourceTracking.ts (hunk 1 above)
Comment (mshanemc) [reply src/sourceTracking.ts:86]: should have been removed.

**Comment by mshanemc** [comment]
closed...all changes are now part of https://github.com/forcedotcom/source-tracking/pull/55

**Pull request closed without merging**